File: dp/largest_palin.cpp
#include <bits/stdc++.h>
using namespace std;
string max(const string a,const string b){
    return a.size() > b.size() ? a:b;
}

// string largest(string s,int i,int j){
//     if(i>j){
//         return "";
//     }
//     if(i==j){
//         return to_string(s[i]);
//     }
//     if(s[i]==s[j]){
//         return s[i]+largest(s,i+1,j-1)+s[j];
//     }else{
//         return max(largest(s,i+1,j),largest(s,i,j-1));
//     }
// }

int main(){
    string s;
    cin>>s;
    // printf("%c ",s[0]);
    // cout<<largest(s,0,s.length()-1)<<endl;
    return 0;
}

File: dp/largest_palin.py
def largest(s,i,j):
    if i>j:
        return 0
    if i==j:
        return str(s[i])
    if s[i]==s[j]:
        return 2+largest(s,i+1,j-1)
    else:
        return max(largest(s,i+1,j),largest(s,i,j-1)) 


s="heflffddslsdse"
print(largest(s,0,len(s)-1))